bundles/core/org.eclipse.smarthome.core.thing/src/main/java/org/eclipse/smarthome/core/thing/internal/ThingManagerImpl.java

```diff
@@ -1258,14 +1251,31 @@ public void setEnabled(ThingUID thingUID, boolean enabled) {
         }
     }
 
+    private void persistThingEnableStatus(ThingUID thingUID, boolean enabled) {
+        if (storage != null) {
+            logger.debug("Cannot persist thing enable status. Storage unavailable.");
+        }
+
+        logger.debug("Thing with UID {} will be persisted as {}.", thingUID, enabled?"enabled.":"disabled.");
+        if (enabled) {
+            // Clear the disabled thing storage. Otherwise the handler will NOT be initialized later.
+            storage.remove(thingUID.getAsString());
+        } else {
+            // Mark the thing as disabled in the storage.
+            storage.put(thingUID.getAsString(), enabled);
+        }
+
+    }
+
     @Override
     public boolean isEnabled(ThingUID thingUID) {
         Thing thing = getThing(thingUID);
 
         if (thing == null) {
```

Comment:
In case the thing is null we should imho also query the database here.

Reply:
Do you think that we have to call one after the other? Isn't it better to only query the database?

Reply:
PR updated!